Zotero/Controllers/PDFWorkerController.swift

```diff
@@ -0,0 +1,203 @@
+                    .disposed(by: disposeBag)
+                return
+            }
+
+            statesByPDFWork[work] = .inProgress
+            setupObserver(for: pdfWorkerWebViewHandler)
+            subject.on(.next(Update(work: work, kind: .inProgress)))
+            switch work.kind {
+            case .recognizer:
+                pdfWorkerWebViewHandler.recognize(file: work.file)
+
+            case .fullText:
+                pdfWorkerWebViewHandler.getFullText(file: work.file)
+            }
+
+            func setupObserver(for pdfWorkerWebViewHandler: PDFWorkerWebViewHandler) {
+                pdfWorkerWebViewHandler.observable
+                    .subscribe(onNext: { [weak self] in
+                        guard let self else { return }
+                        process(result: $0)
+                    })
+                    .disposed(by: disposeBag)
+
+                func process(result: Result<PDFWorkerWebViewHandler.PDFWorkerData, Error>) {
+                    switch result {
+                    case .success(let data):
+                        switch data {
+                        case .recognizerData(let data), .fullText(let data):
+                            cleanupPDFWorker(for: work).subscribe(onSuccess: { $0.on(.next(Update(work: work, kind: .extractedData(data: data)))) })
+                                .disposed(by: disposeBag)
+                        }
+
+                    case .failure(let error):
+                        DDLogError("PDFWorkerController: recognizer failed - \(error)")
+                        cleanupPDFWorker(for: work).subscribe(onSuccess: { $0.on(.next(Update(work: work, kind: .failed))) })
+                            .disposed(by: disposeBag)
+                    }
+                }
+            }
+        }
+    }
+
+    func cancel(work: PDFWork) {
+        DDLogInfo("PDFWorkerController: cancelled \(work)")
+        cleanupPDFWorker(for: work).subscribe(onSuccess: { $0.on(.next(Update(work: work, kind: .cancelled))) })
+            .disposed(by: disposeBag)
+    }
+
+    func cancellAllWorks() {
+        accessQueue.async(flags: .barrier) { [weak self] in
+            guard let self else { return }
+            DDLogInfo("PDFWorkerController: cancel all works")
+            // Immediatelly release all PDFWorker web views.
+            pdfWorkerWebViewHandlersByPDFWork.values.forEach { $0.webViewHandler.removeFromSuperviewAsynchronously() }
+            pdfWorkerWebViewHandlersByPDFWork = [:]
+            // Then cancel actual works, and send cancelled event for each queued work.
+            let works = subjectsByPDFWork.keys + Array(queue.keys)
+            for work in works {
+                cancel(work: work)
+            }
+        }
+    }
+
+    private func cleanupPDFWorker(for work: PDFWork) -> Maybe<PublishSubject<Update>> {
+        return Maybe.create { [weak self] maybe in
```

Comment:
Same as in `RecognizerController`, `Maybe` is not really utilised, basic Swift block should be sufficient and simpler to use.